src/firefly/js/templates/fireflyslate/FireflySlate.jsx

```diff
@@ -104,30 +101,30 @@ export class FireflySlate extends PureComponent {
             return (<div style={{top: 0}} className='loading-mask'/>);
         } else {
             return (
-                <div id='App' className='rootStyle' style={style}>
-                    <header>
-                        <BannerSection {...{menu, appTitle, appIcon, altAppIcon}}/>
-                        <div id={warningDivId} data-decor='full' className='warning-div center'/>
-                        <DropDownContainer
-                            key='dropdown'
-                            footer={footer}
-                            visible={!!visible}
-                            selected={view}
-                            {...{dropdownPanels} } />
-                    </header>
-                    <main style={{height:'100%'}}>
-                        {mainView({expanded, gridView, gridColumns})}
-                    </main>
-                </div>
+                    <div id='App' className='rootStyle' style={style}>
+                        <header>
+                            <BannerSection {...{menu, appTitle, appIcon, altAppIcon}}/>
+                            <div id={warningDivId} data-decor='full' className='warning-div center'/>
+                            <RenderTreeIdCtx.Provider value={{renderTreeId : this.props.renderTreeId}}>
+                                <DropDownContainer
+                                    key='dropdown'
+                                    footer={footer}
+                                    visible={!!visible}
+                                    selected={view}
+                                    {...{dropdownPanels} } />
+                            </RenderTreeIdCtx.Provider>
+                        </header>
+                        <main style={{height:'100%'}}>
+                            <RenderTreeIdCtx.Provider value={{renderTreeId : this.props.renderTreeId}}>
+                                {mainView({expanded, gridView, gridColumns})}
+                            </RenderTreeIdCtx.Provider>
```

Comment:
Yes, i need to change it back to that. I had it on the root but changed it because it was not working, but that was not the problem. I had to break
`RenderTreeIdCtx` into it own file.  I think I was having circular import issues.